From 0a139d7c0f0057ed0d4d97fdbaea007562eee7fd Mon Sep 17 00:00:00 2001
From: Daniel Vetter <daniel.vetter@ffwll.ch>
Date: Tue, 15 Sep 2009 22:57:35 +0200
Patch-mainline: 2.6.33
References: fate#310916
Git-commit: 240a2d12dfff98f8fa1332dc8424284d96f0801e
Subject: [PATCH 0038/2588] drm/i915: fully switch off overlay when not in use

Now that the cache flushing of the memory based overlay regs works,
we can safely switch off the overlay. Beforehand it was only disabled
(like in userspace).

Signed-off-by: Daniel Vetter <daniel.vetter@ffwll.ch>
Signed-off-by: Eric Anholt <eric@anholt.net>
(cherry picked from commit 240a2d12dfff98f8fa1332dc8424284d96f0801e)

Signed-off-by: Takashi Iwai <tiwai@suse.de>
---
 drivers/gpu/drm/i915/intel_overlay.c |    3 +--
 1 files changed, 1 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_overlay.c b/drivers/gpu/drm/i915/intel_overlay.c
index 3f6f3a3..4e88abb 100644
--- a/drivers/gpu/drm/i915/intel_overlay.c
+++ b/drivers/gpu/drm/i915/intel_overlay.c
@@ -323,7 +323,6 @@ static int intel_overlay_off(struct intel_overlay *overlay)
 	}
 
 	/* turn overlay off */
-	/* this is not done in userspace!
 	BEGIN_LP_RING(6);
         OUT_RING(MI_FLUSH);
         OUT_RING(MI_NOOP);
@@ -338,7 +337,7 @@ static int intel_overlay_off(struct intel_overlay *overlay)
 		DRM_ERROR("intel overlay: ring sync failed, hw likely wedged\n");
 		overlay->hw_wedged = 1;
 		return ret;
-	}*/
+	}
 
 	overlay->active = 0;
 
-- 
1.7.6

